### PR TITLE
Use packaging.version, DeprecationWarnings since setuptools 59+ 

### PR DESCRIPTION
**Pull request by Carreau**
This is creating CI failures as we turn warnings into error.
Follow deprecation warning message and use alternative.

See https://setuptools.pypa.io/en/latest/history.html#v59-6-0

Note that this introduces a new dependency.

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
LGTM! I'm fine with new dependency, but i'll wait for others to weigh in before merge

**Comment by tlambert03** [comment]
since the only version string we're ever evaluating is Qt and `platform.release()` ... we might also consider just not using a version dependency at all?  and just using `tuple(str.split('.'))`?

**Comment by Carreau** [comment]
That would be fine with me as well, I just assume that if it was not using a tuple there might be reasons...

**Comment by sofroniewn** [comment]
I'm fine here with either approach. 

> since the only version string we're ever evaluating is Qt and platform.release() ... we might also consider just not using a version dependency at all? and just using tuple(str.split('.'))?

Maybe we just do this as it avoids adding the new dependency, right? Might as well keep things lighter weight

**Comment by sofroniewn** [comment]
I think we can merge this now. Looks like the typing test fail is independent of this PR!

**Pull request merged**
